test-sbt/js/src/main/scala/zio/test/sbt/ZTestTask.scala

```
@@ -0,0 +1,42 @@
+/*
+ * Copyright 2019-2024 John A. De Goes and the ZIO Contributors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package zio.test.sbt
+
+import sbt.testing.{EventHandler, Logger, Task, TaskDef}
+import zio.test.{Summary, TestArgs, ZIOSpecAbstract}
+
+final class ZTestTask(
+  taskDef: TaskDef,
+  spec: ZIOSpecAbstract,
+  runnerType: String,
+  sendSummary: Summary => Unit,
+  testArgs: TestArgs,
+  runtime: zio.Runtime[Any]
+) extends TestTask(
+      taskDef,
+      spec,
+      runnerType,
+      sendSummary,
+      testArgs,
+      runtime
+    ) {
+  // Note: on Scala.js this method is never called - next one is.
+  override def execute(eventHandler: EventHandler, loggers: Array[Logger]): Array[Task] = ???
```

Can you throw an error with a clear message stating the comment, something like:
```
new IllegalStateException("ZTestTask.execute unexpectedly invoked on Scala.js...
```

---

done in https://github.com/zio/zio/pull/10120/commits/eadcaa96d14fefd4948cb113b7c324fcacc5596d